zulip_bots/zulip_bots/lib.py

```diff
@@ -93,24 +114,64 @@ def get(self, key: Text) -> Any:
         self.state_[key] = marshalled_value
         return self.demarshal(marshalled_value)
 
+    @contextmanager
+    def use(self, keys: List[Text]) -> Iterator[BaseStorage]:
```

Comment:
I would remommend `-> Iterator[CachedStorage]:` instead of `-> Iterator[BaseStorage]:`

Reply:
and have only one `BotStorage` and skip `BaseStorage`

Reply:
The second piece of advice is not achievable as I have to set the return type of `use` for the protocol.

Reply:
The dependency of these classes is now `BaseStorage` > `BotStorage` > `CachedStorage` > `StateHandler`. But I think it might not be necessary to have `StateHandler`'s `use` function to return a CachedStorage as the return type because only `BaseStorage` will be visible to the users as they can only see the protocol.

Reply:
Yeah, I understood the circular dependency issue, and `BaseStorage` is probably necessary. Thinking of a work around.

Reply:
I just found that there is a way to define recursive protocol so that we won't need `BaseStorage`.